config/jobs/kubernetes/sig-cluster-lifecycle/k8s-upgrade-gce.yaml

```diff
@@ -0,0 +1,463 @@
+      - --provider=gce
+      - --skew
+      - --test_args=--ginkgo.skip=\[Slow\]|\[Serial\]|\[Disruptive\]|\[Flaky\]|\[Feature:.+\] --kubectl-path=../../../../kubernetes_skew/cluster/kubectl.sh --minStartupPods=8
+      - --timeout=120m
+      - --upgrade_args=--ginkgo.focus=\[Feature:ClusterDowngrade\] --upgrade-target=ci/k8s-stable1 --upgrade-image=gci
+      image: gcr.io/k8s-testimages/kubekins-e2e:v20180716-9145034c9-master
+
+- interval: 2h
+  agent: kubernetes
+  name: ci-kubernetes-e2e-gce-new-master-upgrade-cluster
+  labels:
+    preset-service-account: "true"
+    preset-k8s-ssh: "true"
+  spec:
+    containers:
+    - args:
+      - --timeout=920
+      - --bare
+      - --scenario=kubernetes_e2e
+      - --
+      - --check-leaked-resources
+      - --check-version-skew=false
+      - --env-file=jobs/env/ci-kubernetes-e2e-gce-new-master-upgrade-cluster.env
+      - --extract=ci/latest
+      - --extract=ci/k8s-stable1
+      - --gcp-node-image=gci
+      - --gcp-zone=us-central1-f
+      - --provider=gce
+      - --test_args=--ginkgo.focus=\[Slow\]|\[Serial\]|\[Disruptive\] --ginkgo.skip=\[Flaky\]|\[Feature:.+\] --kubectl-path=../../../../kubernetes_skew/cluster/kubectl.sh --minStartupPods=8
+      - --timeout=900m
+      - --upgrade_args=--ginkgo.focus=\[Feature:ClusterUpgrade\] --upgrade-target=ci/latest
+      image: gcr.io/k8s-testimages/kubekins-e2e:v20180716-9145034c9-master
+
+- interval: 2h
+  agent: kubernetes
+  name: ci-kubernetes-e2e-gce-new-master-upgrade-cluster-new
+  labels:
+    preset-service-account: "true"
+    preset-k8s-ssh: "true"
+  spec:
+    containers:
+    - args:
+      - --timeout=920
+      - --bare
+      - --scenario=kubernetes_e2e
+      - --
+      - --check-leaked-resources
+      - --check-version-skew=false
+      - --env=STORAGE_MEDIA_TYPE=application/vnd.kubernetes.protobuf
+      - --extract=ci/latest
+      - --extract=ci/k8s-stable1
+      - --gcp-node-image=gci
+      - --gcp-zone=us-central1-f
+      - --provider=gce
+      - --skew
+      - --timeout=900m
+      - --upgrade_args=--ginkgo.focus=\[Feature:ClusterUpgrade\] --upgrade-target=ci/latest
+      image: gcr.io/k8s-testimages/kubekins-e2e:v20180716-9145034c9-master
+
+- interval: 2h
+  agent: kubernetes
+  name: ci-kubernetes-e2e-gce-new-master-upgrade-cluster-parallel
+  labels:
+    preset-service-account: "true"
+    preset-k8s-ssh: "true"
+  spec:
+    containers:
+    - args:
+      - --timeout=140
+      - --bare
+      - --scenario=kubernetes_e2e
+      - --
+      - --check-leaked-resources
+      - --check-version-skew=false
+      - --env=STORAGE_MEDIA_TYPE=application/vnd.kubernetes.protobuf
+      # TODO: This is pinned to the 1.8 version of etcd. 1.9 was changed to
+      # 3.1.x. ETCD doesn't downgrade minor versions. So for downgrades, we
+      # pin this to the lower version. The long term fix is to change
+      # downgrade/upgrade to not upgrade/downgrade etcd.
+      - --env=TEST_ETCD_VERSION=3.0.17
```

Comment:
the env files are inlined.

question maybe for @krousey : do we still need to set this? the TODO is a few releases back :-)

Reply:
@jpbetz We pin the etcd version because of downgrade issues. Is downgrading etcd generally supported now? I've noticed issues in the past where ETCD detects that a downgrade is needed, but can't locate the appropriate etcdctl to do the downgrade.

```
2018-07-14 00:00:02 Detecting if migration is needed
Performing etcd 3.2.18 -> 3.1.12 rollback
Starting etcd version 3.2.18 to capture rollback snapshot.
/usr/local/bin/migrate-if-needed.sh: line 200: /usr/local/bin/etcd-3.2.18: not found
/usr/local/bin/migrate-if-needed.sh: line 200: /usr/local/bin/etcdctl-3.2.18: not found
```

from https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-gpu-beta-stable1-cluster-downgrade/39/artifacts/bootstrap-e2e-master/etcd.log

We should at least update the version of etcd we pin to.

Reply:
ok, I'd do this 1:1 port for now to avoid any breaking changes, and address the etcd version as follow ups